### PR TITLE
[FrameworkBundle] Add check for installed yaml component

### LINKED ISSUE #38301: [FrameworkBundle] ConfigDumpReferenceCommand uses yaml even when not present
**Symfony version(s) affected**: 5.1.5 (probably other versions, too)

**Description**  
If you don't use symfony/yaml component, the command "config:dump-reference" is trying to use yaml as default value for "format" option, resulting in an error.

**How to reproduce**  
* create a symfony project without symfony/yaml component (or remove symfony/yaml component from an existing project).
* try to dump configuration for any bundle, e.g. `console config:dump-reference twig`
* see error (in YamlReferenceDumper.php line 111) : 

>Attempted to load class "Inline" from namespace "Symfony\Component\Yaml".  
>Did you forget a "use" statement for another namespace?   

**Possible Solution**  
A check for existance of above class should be added. If not present, default value of "format" option should be "xml" instead of "yaml".

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #38301 
| License       | MIT
| Doc PR        | - 

**Todo:**

- [ ] add tests

### CODE REVIEW

**Review by chalasr — CHANGES_REQUESTED** [review]
The patch makes sense for the case where the developer relies on the default value. 
But IMHO if one explicitly indicates yaml and it's not available, one should get an error suggesting to switch to xml or to install the yaml component. 
This can be achieved by removing the default from the option definition and throwing when needed.

**Comment by fabpot** [comment]
Should be done on master as this is a behavior change.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php [hunk 1]
```diff
@@ -112,6 +113,17 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
 
         $format = $input->getOption('format');
+
+        if ('yaml' === $format && !class_exists(Yaml::class)) {
+            $errorIo->error('Setting the "format" option to "yaml" requires the "Yaml" component. Either install "symfony/yaml" or use "xml".');
+
+            return 1;
+        }
+
+        if (null === $format) {
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: Does that actually work? The `format` option defaults to `yaml`, so we shouldn't get `null` here. Or am I missing something?

What I was thinking about: We could change the default, depending on the availability of the YAML component.

```php
new InputOption('format', null, InputOption::VALUE_REQUIRED, 'The output format (yaml or xml)', class_exists(Yaml::class) ? 'yaml' : 'xml'),
```

Comment (jschaedl) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: > Does that actually work? The format option defaults to yaml, so we shouldn't get null here. Or am I missing something?

Yes, you are right. The plan was to remove the `format` options default value like @chalasr suggested, but somehow I forgot to commit this change.

Comment (jschaedl) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: > What I was thinking about: We could change the default, depending on the availability of the YAML component.

Yes, that's a much better solution. Will change it. Thanks @derrabus

**Review by derrabus — APPROVED** [review]

**Review by fabpot — DISMISSED** [review]

**Comment by fabpot** [comment]
Thinking about it, I think changing the default value depending on the YAML component availability is not a good idea. I would instead keep the default and throw an exception when YAML is used (either explicitly or implicitly) and the component is not installed.

**Comment by derrabus** [comment]
Fair enough. But if we don't change the default, all this PR does is replacing an ugly exception with a nice error message. And that would be a valid thing to do in a bugfix release, wouldn't it?

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php [hunk 2]
```diff
@@ -112,6 +113,13 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->validateConfiguration($extension, $configuration);
 
         $format = $input->getOption('format');
+
+        if ('yaml' === $format && !class_exists(Yaml::class)) {
+            $errorIo->error('Setting the "format" option to "yaml" requires the "Yaml" component. Either install "symfony/yaml" or use "xml".');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: `Setting the "format" option to "yaml" requires the Symfony Yaml component. Try running "composer install "symfony/yaml" or use "--format=xml" instead.`

Comment (jschaedl) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]:
```suggestion
            $errorIo->error('Setting the "format" option to "yaml" requires the Symfony Yaml component. Try running "composer install symfony/yaml" or use "--format=xml" instead.');
```

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php [hunk 3]
```diff
@@ -45,7 +46,7 @@ protected function configure()
             ->setDefinition([
                 new InputArgument('name', InputArgument::OPTIONAL, 'The Bundle name or the extension alias'),
                 new InputArgument('path', InputArgument::OPTIONAL, 'The configuration option path'),
-                new InputOption('format', null, InputOption::VALUE_REQUIRED, 'The output format (yaml or xml)', 'yaml'),
+                new InputOption('format', null, InputOption::VALUE_REQUIRED, 'The output format (yaml or xml)', class_exists(Yaml::class) ? 'yaml' : 'xml'),
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: should be reverted per the discussion

**Comment by nicolas-grekas** [comment]
Thank you @jschaedl.

**Pull request merged**